### PR TITLE
Tests+LibTest: Port to Core::Stream

### PR DESCRIPTION
**Pull request by AtkinsSJ**
As far as I can tell, this is all the tests ported from using Core::File to Core::Stream::File. (Except the IODevice tests of course!)

The test-wasm code is a bit awkward because using TRY on the stream functions returns an ErrorOr that's incompatible with the JS ones. Might be a better way to do that.

### CODE REVIEW

File: Tests/LibWasm/test-wasm.cpp [hunk 1]
```diff
@@ -4,23 +4,30 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
-#include <LibCore/File.h>
+#include <LibCore/Stream.h>
 #include <LibTest/JavaScriptTestRunner.h>
 #include <LibWasm/AbstractMachine/BytecodeInterpreter.h>
 #include <LibWasm/Types.h>
-#include <string.h>
```
Comment (trflynn89) [inline Tests/LibWasm/test-wasm.cpp:11]: `string.h` is still used directly here for `strerror`

Comment (AtkinsSJ) [reply Tests/LibWasm/test-wasm.cpp]: :+1: Thanks

**Comment by trflynn89** [comment]
It seems fine here, but one thing we need to watch out for is that the file sizes we're reading don't exceed 16KB, otherwise `read` doesn't actually read the entire file:
https://github.com/SerenityOS/serenity/pull/12324#discussion_r806023183

We really need to get on making a `read_all` API

**Comment by AtkinsSJ** [comment]
> It seems fine here, but one thing we need to watch out for is that the file sizes we're reading don't exceed 16KB, otherwise `read` doesn't actually read the entire file: [#12324 (comment)](https://github.com/SerenityOS/serenity/pull/12324#discussion_r806023183)
> 
> We really need to get on making a `read_all` API

Good to know, thanks. Previously I was using `read_or_error()` for whole files but it's not really what that's intended for.

**Pull request merged**